packages/x-date-pickers/src/ClockPicker/ClockPicker.tsx

```diff
@@ -262,10 +262,6 @@ export const ClockPicker = React.forwardRef(function ClockPicker<TDate extends u
 
   const isTimeDisabled = React.useCallback(
     (rawValue: number, viewType: ClockPickerView) => {
-      if (date === null) {
```

Comment:
Even if the current date is null we can check if the value is valid, we just can't check the full time.